dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -25,7 +25,10 @@ if (!window.dashboard) {
  */
 dashboard.buildHeader = function (stageData, progressData, currentLevelId, userId, sectionId) {
   stageData = stageData || {};
-  progressData = progressData || {};
+
+  if (progressData === null) {
```

Comment:
Please add a comment explaining that progressData is provided for signed users only.